library/src/androidTest/java/org/xmtp/android/library/TestHelpers.kt

```diff
@@ -41,6 +52,94 @@ class FakeWallet : SigningKey {
         get() = privateKey.walletAddress
 }
 
+class FakeSCWWallet(
+    private val web3j: Web3j,
+    private val credentials: Credentials,
+) : SigningKey {
+    var walletAddress: String = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
+
+    init {
+        runBlocking {
+            createSmartContractWallet()
+        }
+    }
+
+    // Override address to return the created smart contract wallet address
+    override val address: String
+        get() = walletAddress
+
+    override val isSmartContractWallet: Boolean
+        get() = true
+
+    override var chainId: Long = 1L
+
+    companion object {
+        fun generate(
+            web3j: Web3j,
+            credentials: Credentials,
+        ): FakeSCWWallet {
+            return FakeSCWWallet(web3j, credentials).apply {
+                runBlocking { createSmartContractWallet() }
+            }
+        }
+    }
+
+    override suspend fun sign(data: ByteArray): Signature {
+        val smartWallet = CoinbaseSmartWallet.deploy(
+            web3j,
+            credentials,
+            DefaultGasProvider()
+        ).send()
+
+        val randomHash = ByteArray(32)
+        SecureRandom().nextBytes(randomHash)
```

Comment:
Why is the hash random?

This is going through most of the steps to get a real, verifiable, signature...but if it's signing randomness it's not going to verify with a genuine SCW.